File: 剑指offer/24反转链表/24.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
 var reverseList = function(head) {
    if(!head || !head.next)
    {
        return head;
    }
    else{
        let tail = null
        while(head)
        {
            let second = head.next
            head.next = tail 
            tail = head
            head = second
        }
        return tail;
    }
};

File: 剑指offer/24反转链表/24.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        ListNode* tail = NULL, *pre = head;
        while (pre != NULL) {
            ListNode* t = pre->next;
            pre->next = tail;
            tail = pre;
            pre = t;
        }
        return tail;
    }
};